pkg/validation/providers.go

```diff
@@ -48,7 +48,7 @@ func validateProvider(provider options.Provider, providerIDs map[string]struct{}
 
 	// login.gov uses a signed JWT to authenticate, not a client-secret
 	if provider.Type != "login.gov" {
-		if provider.ClientSecret == "" && provider.ClientSecretFile == "" {
+		if provider.ClientSecret == "" && provider.ClientSecretFile == "" && provider.Type != "oidc" {
```

Comment:
OIDC may need client secret, if it uses confidential clients. I think it is optional if PKCE is used (option --code-challenge-method)